src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreWorker.cs

```diff
@@ -159,12 +177,20 @@ private void Reset(bool isDisposing = false)
                 _joinableFactory.Run(
                     async () =>
                     {
-                        using (_joinableCollection.Join())
+                        try
                         {
                             // Do not block VS forever
-                            await Task.WhenAny(_backgroundJobRunner.Value, Task.Delay(TimeSpan.FromSeconds(60)));
+                            // After the specified delay the task will disjoin.
+                            using (_joinableCollection.Join())
```

Comment:
You _shouldn't_ need to join a collection when awaiting an `AsyncLazy<T>`. If you do, you're missing a dependency chain somewhere else.